src/Oscoin/Storage.hs

```diff
@@ -94,7 +94,11 @@ applyTx
 applyTx bs tx = do
     let txHash = Crypto.hash tx
     novel <- isNovelTx bs txHash
-    if | novel     -> Mempool.addTxs [tx] $> Applied [TxAppliedEvent txHash]
+    if | novel     -> do
+            addResult <- Mempool.addTx tx
+            case addResult of
+                Left _  -> pure $ Error []
```

Comment:
It's kind of a shame we are throwing a way the error and not spitting out any telemetry event -- can we add one?